extensions/common/context.cc

```diff
@@ -96,9 +96,8 @@ void extractServiceName(const std::string& host,
 //   host for destination service name.
 void getDestinationService(const std::string& dest_namespace,
                            bool use_host_header, std::string* dest_svc_host,
-                           std::string* dest_svc_name) {
-  std::string cluster_name;
-  getValue({"cluster_name"}, &cluster_name);
+                           std::string* dest_svc_name,
+                           std::string cluster_name) {
```

Comment:
```suggestion
                           const std::string& cluster_name) {
```

Reply:
we are modifying cluster_name.. hence didn't make it a const ref.. and passed it by value rather than copying it to a temp variable... but looks like it's not a good idea from here: https://www.cplusplus.com/articles/z6vU7k9E/

Pass by value when the function does not want to modify the parameter and the
value is easy to copy (ints, doubles, char, bool, etc... simple types. std::string,
std::vector, and all other STL containers are NOT simple types.)

Reply:
Now look at the logic, it seems to be unnecessary to change value of cluster_name. We can just update the logic to:
```
  if (route_name == kBlackHoleRouteName) {
    *dest_svc_name = kBlackHoleCluster;
    return;
  }
  if (route_name == kPassThroughRouteName) {
    *dest_svc_name = kPassThroughCluster;
    return;
  }
  if (cluster_name == kInboundPassthroughClusterIpv4 ||
      cluster_name == kInboundPassthroughClusterIpv6) {
    *dest_svc_name = cluster_name;
    return;
  }
```

Also seems like we are getting route name here as well, so we are fetching it twice as well. Can we pass it in instead?